tests/test_toolbox.py

```diff
@@ -165,6 +172,12 @@ def test_indsFromXrange(self):
         foo = xrange(23, 39)
         self.assertEqual([23, 39], tb.indsFromXrange(foo))
 
+    @unittest.expectedFailure
+    def test_indsFromXrange_zerolen(self):
+        """indsFromXrange claims useful when zero len range, seems not true"""
+        foo = xrange(20, 10)  # empty
+        self.assertEqual([23, 39], tb.indsFromXrange(foo))
```

Comment:
Why do you expect this value? I get:
```python
In [5]: spacepy.toolbox.indsFromXrange(xrange(20, 10))
Out[5]: [20, 20]
```
which is eminently reasonable.

Reply:
I get the useless
```
In [2]: import spacepy.toolbox as tb
In [3]: tb.indsFromXrange(range(20, 10))
Out[3]: [20, 10]
```

out of python3, maybe it should error with `NotImplementedError` or something in py3.

Reply:
We should check that then; assuming the py2k result is "right" then fix on py3k.

Reply:
Still not sure why you expect [23, 39] though.

Reply:
I'm honestly not sure about the utility of this function; it doesn't work with a specified stride, for instance. There's a definite difference between the py2k and py3k behaviour in the way the `str` is rendered; my inclination is let's just slap this in as a regression test, have separate expected for 2 and 3, and worry about the defined correct behaviour later.

Reply:
This was added by @drsteve in f725534f1d5a256b52d648ae7e5d6f8d541763fb to support omni. Basically it was using tOverlapHalf to find a start/end index, but since that was returned as an xrange, then omni wanted to derive the indices from it, and since it wanted to extend past the start/finish, it couldn't just cast to a list and get an empty list. The situation it was written for was only for things of form xrange(20, 20) (from which it would go grabbing 19:22), never with the second one smaller than the first.

Let's just document this as undefined if end < start or if there's a stride and call it good.

Reply:
> Still not sure why you expect [23, 39] though.

Yeah I have no idea either...